seacrowd/sea_datasets/indoner_tourism/indoner_tourism.py

```diff
@@ -0,0 +1,196 @@
+    B-LOC (3): Beginning of a location entity.
+    I-LOC (4): Continuation of a location entity.
+    B-FAS (5): Beginning of a facility entity.
+    I-FAS (6): Continuation of a facility entity.
+"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="indoner_tourism_source",
+            version=SOURCE_VERSION,
+            description="indoner_tourism source schema",
+            schema="source",
+            subset_id="indoner_tourism",
+        ),
+        SEACrowdConfig(
+            name="indoner_tourism_seacrowd_seq_label",
+            version=SEACROWD_VERSION,
+            description="indoner_tourism SEACrowd schema",
+            schema="seacrowd_seq_label",
+            subset_id="indoner_tourism",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "indoner_tourism_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    'tokens'  : datasets.Sequence(datasets.Value("string")),
+                    'ner_tags': datasets.Sequence(
+                                    datasets.ClassLabel(names=["O", "B-WIS", "I-WIS", "B-LOC", "I-LOC", "B-FAS", "I-FAS"])
+                                ),
+                }
+            )
+        elif self.config.schema == "seacrowd_seq_label":
+            features = schemas.seq_label.features(["O", "B-WIS", "I-WIS", "B-LOC", "I-LOC", "B-FAS", "I-FAS"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS[_DATASETNAME]
+        path = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": path,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        tokens = []
+        ner_tags = []
+        counter = 0
+        with open(filepath, encoding="utf-8") as file:
+            for line in file:
+                if line.strip() == "":
+                    if self.config.schema == "source":
+                        yield(
+                            counter,
+                            {
+                                'tokens'  : tokens,
+                                'ner_tags': ner_tags,
+                            }
+                        )
```

Comment:
The use of `yield` here seems non-idiomatic. Perhaps we can do something like this:

```python
yield counter, {"tokens": tokens, "ner_tags": ner_tags}
```

Same comment in lines 177--184

Reply:
Resolved!